python/tank/descriptor/io_descriptor/base.py

```diff
@@ -474,6 +471,12 @@ def get_dict(self):
         """
         return self._descriptor_dict
 
+    def get_type(self):
```

Comment:
can we get rid of this? why is it needed beyond introspecting the instances returned by the factory?